cmd/soroban-rpc/internal/db/ledger.go

```diff
@@ -30,12 +32,27 @@ func NewLedgerReader(db *DB) LedgerReader {
 	return ledgerReader{db: db}
 }
 
-// GetAllLedgers returns all ledgers in the database.
-func (r ledgerReader) GetAllLedgers(ctx context.Context) ([]xdr.LedgerCloseMeta, error) {
-	var results []xdr.LedgerCloseMeta
+// StreamAllLedgers runs f over all the ledgers in the database (until f errors or signals it's done).
+func (r ledgerReader) StreamAllLedgers(ctx context.Context, f StreamLedgerFn) error {
 	sql := sq.Select("meta").From(ledgerCloseMetaTableName).OrderBy("sequence asc")
-	err := r.db.Select(ctx, &results, sql)
-	return results, err
+	q, err := r.db.Query(ctx, sql)
+	if err != nil {
+		return err
+	}
+	var closeMeta xdr.LedgerCloseMeta
```

Comment:
could you move this line `var closeMeta xdr.LedgerCloseMeta` into the for-loop to ensure the content of the `closeMeta` variable is always the default value ?

Reply:
Uhm, I think it’s fine, because the data structure is deserialized in full.

Reply:
ok

Reply:
I did it anyways since I don't think it gave us more performance.